erpnext/accounts/report/item_wise_purchase_register/item_wise_purchase_register.py

```diff
@@ -53,8 +53,8 @@ def _execute(filters=None, additional_table_columns=None, additional_query_colum
 
 		row = {
 			'item_code': d.item_code,
-			'item_name': item_record.item_name,
-			'item_group': item_record.item_group,
+			'item_name': item_record.item_name if item_record else d.item_name,
+			'item_group': item_record.item_group if item_record else d.item_group,
```

Comment:
@pateljannat where do such cases occur where the user may not enter Item Code? (curious)
This field (item code in Purchase Invoice) is not kept mandatory because it could be blank only in the case if it was created from **Opening Invoice Creation Tool**.
In which case you will see **Is Opening** set to yes in the Invoice:
![Screenshot 2021-01-20 at 5 16 44 PM](https://user-images.githubusercontent.com/25857446/105170478-fd441a00-5b42-11eb-8a2c-a6dadf7026f3.png)

The fix looks alright, but should a case like this occur manually at all ? And opening invoices aren't item purchases so they shouldn't even be fetched in this report, right ?

Reply:
@marination I encountered such a scenario through a support issue. The customer had 3 such invoices where there was no item code. Also, they had **Is Opening** set as No. The item names mentioned were Labour outsource or Freight Charges. So I think the customer is using this for scenarios wherein its not actually an item but rather a service. 

As you mentioned that this field was not kept mandatory because it could be blank only in the case if it was created from Opening Invoice Creation Tool, then maybe there should be a validation that triggers at the submission of the invoice itself? I had discussed this issue with Rohit before making the fix and he mentioned that creation of invoices without item code is allowed. Please let me know your thoughts.